server/src/main/java/keywhiz/Tracing.java

```diff
@@ -0,0 +1,91 @@
+package keywhiz;
+
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.log.Fields;
+import io.opentracing.tag.Tags;
+import io.opentracing.util.GlobalTracer;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public final class Tracing {
+  private Tracing() {}
+
+  public static <T> T trace(String spanName, Supplier<T> delegate) {
+    Tracer tracer = GlobalTracer.get();
+
+    Span span = tracer.buildSpan(spanName).start();
+    try {
+      try (Scope ignored = tracer.scopeManager().activate(span)) {
+        return delegate.get();
+      } catch (Exception e) {
+        tagErrors(span, e);
+        throw e;
+      }
+    } finally {
+      span.finish();
+    }
+  }
+
+  public static <T> T tagErrors(Supplier<T> delegate) {
+    Span span = GlobalTracer.get().activeSpan();
+    try {
+      return delegate.get();
+    } catch (Exception e) {
+      tagErrors(span, e);
+      throw e;
+    }
+  }
+
+  private static void tagErrors(Span span, Exception e) {
+    span.setTag(Tags.ERROR, true);
+
+    tagErrorsOpenTracing(span, e);
+    tagErrorsDataDog(span, e);
+  }
```

Comment:
I'm wary of introducing any mention of DataDog (let alone a dependency, which would grant us access to @Trace and make this helper code obsolete), but, it's popular and that's what we use, and using their specific tags results in a much better experience on our end.

Reply:
What's the issue with mentioning DataDog? Is it that we want to avoid being tied to a specific library/project?

Reply:
Yeah, I look at DataDog as a Square-specific dependency whereas Keywhiz is supposed to be more generic due to its open-sourciness. OTOH I don't want to unnecessarily handicap ourselves.